services/report-charts/src/weaver_report/charts/nap.py

```
@@ -0,0 +1,266 @@
+from ..io import Run
+
+
+# ---------------------------------------------------------------------------
+# Helpers — extract nap / sleep records from the trace
+# ---------------------------------------------------------------------------
+
+
+def _nap_records(run: Run) -> list[dict[str, Any]]:
+    """One dict per `context_nap` span, with the attributes flattened."""
+    if run.trace_path is None:
+        raise ChartUnavailable("no trace file — nap analysis requires the JSONL")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)